server/src/server.ts

```diff
@@ -0,0 +1,151 @@
+// Also include all preview / proposed LSP features.
+let connection = createConnection(ProposedFeatures.all)
+
+// Create a simple text document manager. The text document manager
+// supports full document sync only.
+let documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument)
+
+let hasConfigurationCapability: boolean = false
+let hasWorkspaceFolderCapability: boolean = false
+let hasDiagnosticRelatedInformationCapability: boolean = false
+
+connection.onInitialize(async (params: InitializeParams) => {
+  let capabilities = params.capabilities
+
+  // Does the client support the `workspace/configuration` request?
+  // If not, we will fall back using global settings.
+  hasConfigurationCapability = !!(
+    capabilities.workspace && !!capabilities.workspace.configuration
+  )
+  hasWorkspaceFolderCapability = !!(
+    capabilities.workspace && !!capabilities.workspace.workspaceFolders
+  )
+  hasDiagnosticRelatedInformationCapability = !!(
+    capabilities.textDocument &&
+    capabilities.textDocument.publishDiagnostics &&
+    capabilities.textDocument.publishDiagnostics.relatedInformation
+  )
+
+  const sdkQueryEnginePath = await util.getSdkQueryEnginePath()
+  if (!fs.existsSync(sdkQueryEnginePath)) {
+    try {
+      const sdkDir = path.dirname(require.resolve('@prisma/sdk/package.json'))
+      await download({
+        binaries: {
+          'query-engine': sdkDir,
+        },
+        failSilent: false,
+      })
+    } catch (err) {
+      // No error on install error.
+      connection.console.error('Cannot install prisma query-engine: ' + err)
+    }
+  }
+
+  const binPathPrismaFmt = await util.getBinPath()
+  if (!fs.existsSync(binPathPrismaFmt)) {
+    try {
+      await install(binPathPrismaFmt)
+      connection.console.info(
+        'Prisma plugin prisma-fmt installation succeeded.',
+      )
+    } catch (err) {
+      // No error on install error.
+      connection.console.error('Cannot install prisma-fmt: ' + err)
+    }
+  }
+
+  const result: InitializeResult = {
+    capabilities: {
+      definitionProvider: true,
+      documentFormattingProvider: true,
+      /* // Tell the client that the server supports code completion
+      completionProvider: {
+        resolveProvider: true,
+        triggerCharacters: ['@']
+      } */
+    },
+  }
+
+  return result
+})
+
+const messageHandler = new MessageHandler()
+
+connection.onDocumentFormatting(params =>
+  messageHandler.handleDocumentFormatting(params, documents),
+)
+
+documents.onDidChangeContent(change => {
+  validateTextDocument(change.document)
+})
+
+documents.onDidOpen(open => {
+  validateTextDocument(open.document)
+})
+
+async function validateTextDocument(textDocument: TextDocument): Promise<void> {
+  const text = textDocument.getText(fullDocumentRange(textDocument))
+  const binPath = await util.getBinPath()
+  const res = await lint(binPath, text)
+  let diagnostics: Diagnostic[] = []
```

Comment:
```suggestion
  const diagnostics: Diagnostic[] = []
```
Really small detail, I try to use `const` everywhere I can, here it would only protects the reference for array and object so you can still change the content of them